bitcoin/src/bip32.rs

```diff
@@ -607,14 +607,11 @@ impl ExtendedPrivKey {
             return Err(Error::WrongExtendedKeyLength(data.len()));
         }
 
-        let network = if data[0..4] == [0x04u8, 0x88, 0xAD, 0xE4] {
-            Network::Bitcoin
-        } else if data[0..4] == [0x04u8, 0x35, 0x83, 0x94] {
-            Network::Testnet
-        } else {
-            let mut ver = [0u8; 4];
-            ver.copy_from_slice(&data[0..4]);
-            return Err(Error::UnknownVersion(ver));
+        let network = match data {
+            [0x04u8, 0x88, 0xAD, 0xE4, ..] => Network::Bitcoin,
+            [0x04u8, 0x35, 0x83, 0x94, ..] => Network::Testnet,
+            [b0, b1, b2, b3, ..] => return Err(Error::UnknownVersion([*b0, *b1, *b2, *b3])),
```

Comment:
Cool trick I recently learned:
```suggestion
            [b0, b1, b2, b3, ..] => Err(Error::UnknownVersion([*b0, *b1, *b2, *b3]))?,
```

Reply:
Heh, I think that's a little too clever.

Reply:
Even though I know that the condition inside will get elided it still feels wrong...

Reply:
Fair points, I won't use it. I have a feeling I may have done so already - I've been using it for a week or so.

Reply:
Heh, it's not in the current codebase fwiw.

Reply:
Cool, thanks.